### DISCUSSION TITLE
8.2.0-beta OpenAI Assistant API support

### DISCUSSION THREAD

**Discussion by kayhantolga** [betalgo/openai#544] [Announcements]
Assistant feature!

https://www.nuget.org/packages/Betalgo.OpenAI/8.2.0-beta
https://github.com/betalgo/openai/releases/tag/v8.2.0-beta

I'm thrilled to finally release the beta version of the long-awaited Assistant feature. I know, I'm a bit late, but believe me, I've been doing my best to deliver this feature to you in the best possible way. Although I couldn't include all the features yet, many of the main functions are ready to use.
I have a request for you - please try out this beta version and share your feedback with me. Your opinions are incredibly valuable to me because they'll help me further improve the Assistant feature and turn it into a tool that meets your expectations.

Happy coding, everyone!
Tolga

**Comment by onerkaya** [comment]
Unfortunately, the https://github.com/betalgo/openai/wiki/Function-Calling example is available as tools in the beta. It does not exist as Functions. The examples given as Tools produce errors.

'type' is a required property - 'tools.0'

Also, can netstandard2.0 support be added to the Betalgo.OpenAI.Utilities library?

**Reply by kayhantolga** [reply]
Please check playground, there are some samples. It is pretty easy to migrate from functions to tools.

**Reply by kayhantolga** [reply]
I am not planning to support netstandard2.0 unless I see that there is demand. I believe that very few projects will be using it. I may consider support after I finish other missing features, if there is a demand for it.

**Comment by Nicelittlewood** [comment]
Assistant works pretty well!! But it missing Streams support :(

@kayhantolga
Are Assistant' streams already in development or is something that will come later?

thanks for all!

<3

**Reply by kayhantolga** [reply]
Thanks, Stream will be supported after all features are covered https://github.com/betalgo/openai/wiki/Feature-Availability. I am actively working on it.

**Reply by Nicelittlewood** [reply]
> Thanks, Stream will be supported after all features are covered https://github.com/betalgo/openai/wiki/Feature-Availability. I am actively working on it.

May you could tell to my boss, please? XD

TY!

**Comment by pappde** [comment]
@kayhantolga Thanks for the release. Works great. Streams will be nice, but in the meantime, RunRetrieve() polling works fine.

I would like to access the run steps, and can see that while the endpoint is there, the Response classes and RunService calls are not. Should I add in a fork and submit a pull request? 

UPDATE: see https://github.com/betalgo/openai/pull/545

**Reply by kayhantolga** [reply]
Thanks @pappde, I got help from your PR, but my local changes were causing a merge conflict. It was easier to continue with my local files.

**Comment by Nicelittlewood** [comment]
@kayhantolga probably found a bug, but i need confirmation that i'm not missing something:

Everytime i create a message with MessageCreate, it will create an empty "vectore store"

```c#
	var messageResult = openAiService.Messages.MessageCreate(thread.Id, new MessageCreateRequest
	{
		Role = StaticValues.AssistantsStatics.MessageStatics.Roles.User,
		Content = _message,
		FileIds = null
	}).Result;

``` 

![image](https://github.com/betalgo/openai/assets/4989442/607b77e0-dec1-45f7-af5a-15abe73e301a)


as you see i tried to pass Null to FileIds, but it creates everytime in anycase. This not happen in playground so i think is something releated to the API.

**Reply by kayhantolga** [reply]
I will look into this. Is it causing any additional charges on the bill, or just spamming from the vector store?

**Comment by kayhantolga** [comment]
https://github.com/betalgo/openai/discussions/547 => 8.2.1
